ipn/ipnserver/server.go

```diff
@@ -613,9 +613,18 @@ func Run(ctx context.Context, logf logger.Logf, logid string, getEngine func() (
 
 	var store ipn.StateStore
 	if opts.StatePath != "" {
-		store, err = ipn.NewFileStore(opts.StatePath)
-		if err != nil {
-			return fmt.Errorf("ipn.NewFileStore(%q): %v", opts.StatePath, err)
+		const kubePrefix = "kube:"
+		switch {
+		case strings.HasPrefix(opts.StatePath, kubePrefix):
+			store, err = ipn.NewKubeStore(opts.StatePath[len(kubePrefix):])
```

Comment:
I'd just use strings.TrimPrefix for clarity/consistency with the line above.

It'll be imperceptible slower but won't even allocate.

Reply:
Also, assign it to a variable so it's clear what the suffix _is_

```go
    dingus := strings.TrimPrefix(opts.StatePath, kubePrefix)
    store, err := ipn.NewKubeStore(dingus)
```